File: brainbox-gateway/interfaces/image.py
from core.logger import setup_logger
from core.config import settings
from httpx import AsyncClient, Timeout, Limits, AsyncHTTPTransport

logger = setup_logger("interfaces/image.py")

class ImageInterface:
    def __init__(self):
        self.base_url = f"{settings.NEURAL_SERVICE_URL}/image"
        self.client = AsyncClient(
            timeout=Timeout(60.0),
            limits=Limits(max_connections=100),
            transport=AsyncHTTPTransport(retries=3)
        )

    async def generate_answer(self, headers: dict, text: str):
        try:
            logger.info("Получен /generate-answer запрос для сервиса нейросетей!")
            return await self.client.post(
                url=f"{self.base_url}/generate-answer",
                headers=headers,
                json={"text": text}
            )
        except Exception as e:
            logger.error(f"Ошибка генерации ответа: {str(e)}", exc_info=True)
            raise

    async def view_image(self, headers: dict, image_uid: str):
        try:
            logger.info("Получен /view/{image_uid} запрос для сервис нейросетей!")
            return await self.client.get(
                url=f"{self.base_url}/view/{image_uid}",
                headers=headers
            )
        except Exception as e:
            logger.error(f"Ошибка отображения картинки: {str(e)}", exc_info=True)
            raise

    async def download_image(self, headers: dict, image_uid: str):
        try:
            logger.info("Получен /download/{image_uid} запрос для сервиса нейросетей!")
            return await self.client.get(
                url=f"{self.base_url}/download/{image_uid}",
                headers=headers
            )
        except Exception as e:
            logger.error(f"Ошибка скачивания картинки: {str(e)}", exc_info=True)
            raise

    async def get_image_messages(self, headers: dict):
        try:
            logger.info("Получен /get-image-messages запрос для сервиса нейросетей!")
            return await self.client.get(
                url=f"{self.base_url}/get-image-messages",
                headers=headers
            )
        except Exception as e:
            logger.error(f"Ошибка получения истории чата: {str(e)}", exc_info=True)
            raise

    async def delete_image_messages(self, headers: dict):
        try:
            logger.info("Получен /delete-image-messages запрос для сервиса нейросетей!")
            return await self.client.delete(
                url=f"{self.base_url}/delete-image-messages",
                headers=headers
            )
        except Exception as e:
            logger.error(f"Ошибка удаления истории чата: {str(e)}", exc_info=True)
            raise

image_interface = ImageInterface()

File: brainbox-client/src/utils/api/image.js
import axios from "axios";

const GATEWAY_URL_IMAGE = window.appConfig.GATEWAY_URL_IMAGE;

axios.defaults.withCredentials = true;

export const generateAnswer = async (text) => {
  try {
    const response = await axios.post(`${GATEWAY_URL_IMAGE}/generate-answer`, {
      text: text,
    });
    return response.data;
  } catch (error) {
    console.error("Generate answer error: ", error);
    throw error;
  }
};

export const checkTaskStatus = async (task_id) => {
  try {
    const response = await axios.get(`${GATEWAY_URL_IMAGE}/tasks/${task_id}`);
    return response.data;
  } catch (error) {
    console.error("Check task status error: ", error);
    throw error;
  }
};

export const viewImage = async (image_uid) => {
  try {
    const response = await axios.get(`${GATEWAY_URL_IMAGE}/view/${image_uid}`, {
      responseType: "blob",
      withCredentials: true,
    });
    return URL.createObjectURL(response.data);
  } catch (error) {
    console.error("View image error: ", error);
  }
};

export const downloadImage = async (image_uid) => {
  try {
    const response = await axios.get(
      `${GATEWAY_URL_IMAGE}/download/${image_uid}`,
      {
        responseType: "blob",
        withCredentials: true,
      }
    );
    const url = window.URL.createObjectURL(new Blob([response.data]));
    const link = document.createElement("a");
    link.href = url;
    link.setAttribute("download", `image_${image_uid}.jpg`);
    document.body.appendChild(link);
    link.click();
    link.remove();
    window.URL.revokeObjectURL(url);
  } catch (error) {
    console.error("Download image error: ", error);
    throw error;
  }
};

export const getImageMessages = async () => {
  try {
    const response = await axios.get(
      `${GATEWAY_URL_IMAGE}/get-image-messages`,
      {
        withCredentials: true,
        headers: {
          "Content-Type": "application/json",
        },
      }
    );
    return {
      messages: response.data.messages.map((msg) => ({
        ...msg,
        table: "image_chat",
      })),
    };
  } catch (error) {
    console.error("Get image messages error: ", error);
    throw error;
  }
};

export const deleteImageMessages = async () => {
  try {
    const response = await axios.delete(
      `${GATEWAY_URL_IMAGE}/delete-image-messages`,
      {
        withCredentials: true,
        headers: {
          "Content-Type": "application/json",
        },
      }
    );
    return response.data;
  } catch (error) {
    console.error("Delete image messages error: ", error);
  }
};
